app/src/main/java/org/feature/fox/coffee_counter/data/local/Item.kt

```diff
@@ -0,0 +1,16 @@
+package org.feature.fox.coffee_counter.data.local
+
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+
+@Entity(tableName = "items")
+data class Item(
+    @PrimaryKey
+    val id: String,
+
+    var name: String,
+
+    var amount: Int? = 0,
```

Comment:
Why is the amount declared as nullable?
If you are trying to implement the `infinite`-amount behaviour try avoiding `null`, since Kotlin is based on null-safety.
It might be better to use `-1` for the represenation of the `infinite`-amount.

Reply:
There is not just infinite-amount, but also an amount of zero, especially because the amount field is optional. It makes more sense to interpret an empty amount as 0 instead of interpreting it as infinite.

Reply:
I was thinking of:
`> 0` = An amount of items is set.
`= 0` = 0 Items are left. It is not possible to buy this item anymore.
`< 0` = -1 stands for "you didn't specify the amount"

I think that what you did with = 0 as an initializer makes totally sense.
My question was more regarding the `?`-Operator. Is there a reason why the amount is declared as nullable? Because if you initialize it as 0 i should never be `null`.